GUI/UI/UICharacter.cpp

```diff
@@ -948,141 +939,60 @@ void CharacterUI_DrawPaperdoll(Player *player) {
-    if (IsPlayerWearingWatersuit[uPlayerID]) {  // акваланг
-        render->DrawTextureAlphaNew(pPaperdoll_BodyX / 640.0f,
-                                    pPaperdoll_BodyY / 480.0f,
-                                    papredoll_dbods[uPlayerID - 1]);
+    render->DrawTextureAlphaNew(467 / 640.0f, 0, ui_character_inventory_paperdoll_background);
+
+    ItemGen *itemMainHand = player->GetMainHandItem();
+    ItemGen *itemOffHand = player->GetOffHandItem();
+    bool bTwoHandedGrip = itemMainHand && (itemMainHand->GetItemEquipType() == EQUIP_TWO_HANDED || itemMainHand->GetPlayerSkillType() == PLAYER_SKILL_SPEAR && !itemOffHand);
+
+    // Aqua-Lung
+    if (IsPlayerWearingWatersuit[uPlayerID]) {
+        render->DrawTextureAlphaNew(pPaperdoll_BodyX / 640.0f, pPaperdoll_BodyY / 480.0f, paperdoll_dbods[uPlayerID - 1]);
         if (!bRingsShownInCharScreen)
-            render->ZDrawTextureAlpha(
-                pPaperdoll_BodyX / 640.0f, pPaperdoll_BodyY / 480.0f,
-                papredoll_dbods[uPlayerID - 1], player->pEquipment.uArmor);
-
-        //Рука не занята или ...
-        if (!player->GetItem(&PlayerEquipment::uMainHand) ||
-            (player->GetMainHandItem()->GetItemEquipType() !=
-             EQUIP_TWO_HANDED) &&
-                (player->GetMainHandItem()->GetItemEquipType() !=
-                     PLAYER_SKILL_SPEAR ||
-                 player->GetItem(&PlayerEquipment::uShield)))
-            render->DrawTextureAlphaNew(
-                (pPaperdoll_BodyX + pPaperdoll_LeftHand[pBodyComplection][0]) /
-                    640.0f,
-                (pPaperdoll_BodyY + pPaperdoll_LeftHand[pBodyComplection][1]) /
-                    480.0f,
-                papredoll_dlads[uPlayerID - 1]);
-        // -----------------------------------------------------(Hand/Рука)---------------------------------------------------------------
-        if (player->GetItem(&PlayerEquipment::uMainHand)) {
-            item = player->GetMainHandItem();
-            item_X = pPaperdoll_BodyX +
-                     paperdoll_Weapon[pBodyComplection][1][0] -
-                     pItemsTable->pItems[item->uItemID].uEquipX;
-            item_Y = pPaperdoll_BodyY +
-                     paperdoll_Weapon[pBodyComplection][1][1] -
-                     pItemsTable->pItems[item->uItemID].uEquipY;
+            render->ZDrawTextureAlpha(pPaperdoll_BodyX / 640.0f, pPaperdoll_BodyY / 480.0f, paperdoll_dbods[uPlayerID - 1], player->pEquipment.uArmor);
+
+        // hands aren't in two handed grip pose
+        if (!bTwoHandedGrip) {
+            item_X = pPaperdoll_BodyX + pPaperdoll_LeftHand[pBodyComplection][0];
+            item_Y = pPaperdoll_BodyY + pPaperdoll_LeftHand[pBodyComplection][1];
+
+            render->DrawTextureAlphaNew(item_X / 640.0f, item_Y / 480.0f, paperdoll_dlads[uPlayerID - 1]);
+        }
+
+        // main hand's item
+        if (item = itemMainHand) {
```

Comment:
This will be a source of confusion down the line. I suggest writing it this way:

item = itemMainHand;
if (item) {
    // do something with item.
}